### PR TITLE
259 route view form

### LINKED ISSUE #259: routeView (form)
How to open:
* Click routeId at /routes/ list

Two tabs: Reitin tiedot | Reitin suunnat

* Use generic tabs component

Reitin suunnat tab content has the old route component from /routes/ view

Reitin tiedot has a route form

### PR DESCRIPTION
**Pull request by Divinare**
Closes #259 

BE PR: https://gitlab.hsl.fi/jore/jore-map-backend/merge_requests/52

* new route, edit route
* UI: displays if routeId is already taken
* capitalized route ID (+ also capitalized lineId in lineForm)
* route/new & line/new, disabled clicking of "Reitit" tab (that caused a bug and user doesn't have to click that when creating new line/route)
* new card: #840 to finish routePathTab in routeForm
* routeId (reitunnus) validation will be done here: #821

### CODE REVIEW

File: src/components/sidebar/routeView/RouteInfoTab.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    invalidPropertiesMap: object;
+    setValidatorResult: (property: string, validationResult: IValidationResult) => void;
+}
+
+@inject('routeStore', 'errorStore')
+@observer
+class RouteInfoTab extends React.Component<IRouteInfoTabProps, IRouteInfoTabState>{
+    private existinRouteIds: string[] = [];
+
+    constructor(props: any) {
```
Comment (kronken) [inline src/components/sidebar/routeView/RouteInfoTab.tsx]: props: IRouteInfoTabProps

File: src/components/sidebar/routeView/RoutePathTab.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { inject, observer } from 'mobx-react';
+import { RouteStore } from '~/stores/routeStore';
+import * as s from './routePathTab.scss';
```
Comment (kronken) [inline src/components/sidebar/routeView/RoutePathTab.tsx]: Import order

File: src/components/sidebar/routeView/RoutePathTab.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+interface IRoutePathTabProps {
+    routeStore?: RouteStore;
+}
+
+@inject('routeStore', 'errorStore')
+@observer
+class RoutePathTab extends React.Component<IRoutePathTabProps, IRoutePathTabState>{
+    constructor(props: any) {
```
Comment (kronken) [inline src/components/sidebar/routeView/RoutePathTab.tsx]: props: IRoutePathTabProps

File: src/services/routeService.ts [hunk 4]
```diff
@@ -22,13 +25,36 @@ class RouteService {
         return routes.map((res: IRoute) => res!);
     }
 
+    /**
+     * @return {string[]} list of all routeIds
+     */
+    public static fetchAllRoutes = async (): Promise<string[]> => {
```
Comment (kronken) [inline src/services/routeService.ts]: fetchAllRouteIds

**Comment by Divinare** [comment]
Added some description to task #846 where links to line / routeView will be added

**Comment by Divinare** [comment]
Review fixes done.

**Pull request merged**